### DISCUSSION TITLE
htmx.trigger usage

### DISCUSSION THREAD

**Discussion by aurelian2020** [bigskysoftware/htmx#2857] [Q&A] [answered]
Hello,

When user clicks on an a-href link:

`<a href='javascript:void(0);' onClick='doMore()'>click here</a>`

I need to trigger the changed event on this input:

`<input type="search" name="search" hx-post="process.php" hx-trigger="input changed delay:500ms, search" hx-target="#search-results">`

In the javascript function doMore() which executes when click on the link, I try to use:

`htmx.trigger(...)`

but can't make it work.

Please help.

**Comment by Telroshan** [comment]
Hey, 

First of all, please include exactly what you've tried when reporting issues. Sometimes the issue are the parameters themselves so it always helps to see what combinations you tried exactly.

I'm just gut guessing here based on what you've written, but I notice you said
>  trigger the changed event 

Though, you're not using a [`change` event](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/change_event), but an [input event](https://developer.mozilla.org/en-US/docs/Web/API/Element/input_event) with the htmx trigger's ["changed" modifier](https://htmx.org/attributes/hx-trigger/).

What your `hx-trigger` specs says here, is that the request should be fired whenever an `input` event fires and that the input's value changed from last time.
I would guess that if you simply trigger an `input` event using `htmx.trigger` in this case, and the value didn't change, htmx will simply ignore it.

You could either instead trigger a `search` event with `htmx.trigger` considering that you've already specified `search` with no modifiers in your `hx-trigger`, or maybe even add a custom event and trigger this one.

Hope this helps!

**Reply by aurelian2020** [reply]
Thank you so much for your answer, I really apreciate it!

I tried this:

```
var element = document.querySelectorAll('[hx-post="process.php"]')[0];
element.value = clicked_search_term;
htmx.trigger(element, search);
```
It does not work. In browser Developer Tools - Console there is an error: /search is not defined/ - at the line with htmx.trigger

What I am trying to achieve is this:
1. I already have a working search input box:

`<input type="search" name="search" hx-post="process.php" hx-trigger="input changed delay:500ms, search" hx-target="#search-results">`

2. I have a link in another div in the page and I want this: when user clicks on this link, it should trigger the search box because I change the content of the search box before issuing htmx.trigger(...) .

This is a small functionality; in the page I have many other functionalities not related to this. Posting the whole page it would be too complicated and, probably, useless.

Maybe I don't understand how this htmx.trigger it's supposed to work...

I don't want to write another code to duplicate the search functionality. Can I do this with htmx.trigger?

**Reply by Telroshan** [reply, marked as answer]
> It does not work. In browser Developer Tools - Console there is an error: /search is not defined/ - at the line with htmx.trigger

Looking at your example above, it seems like you pass `search` directly, as if it was a variable, but htmx.trigger [expects an event name](https://htmx.org/api/#trigger) here, thus a string

Could you try `htmx.trigger(element, "search");` instead and let met know how it goes?

**Reply by aurelian2020** [reply]
It does the job! 
Thank you so much, I really appreciate it!
